### ISSUE TITLE
Bard field error: Arr::isAssoc() must be of the type array (since Statamic 3.0.4)

### ISSUE THREAD

**Issue by jimblue** [statamic/cms#2369]
## Bug Description

After updating to Statamic release 3.0.5 bard page with bard field return an error.

## Extra Detail

Using bard field like suggested in the doc:

```twig
<div itemprop="articleBody">
  {{ bard_field }}
  {{ partial src="templates/blog/post/sets/{type}" }}
  {{ /bard_field }}
</div>
```

The error:

<img width="1310" alt="Screenshot 2020-09-05 at 23 19 55" src="https://user-images.githubusercontent.com/10623367/92313648-5feeac00-efce-11ea-9819-6cdb761cd666.png">

## Environment

```bash
Statamic 3.0.5 Pro
Laravel 7.27.0
PHP 7.4.10
4rn0/statamic-v3-image-optimizer 1.0.2
jimblue/htmlmin 1.0.0
jimblue/instagram dev-develop
jimblue/mapbox 1.0.0
jimblue/webpacker dev-develop
statamic/seo-pro 2.0.10
```

**Comment by vitor-ao** [comment]
Same error happens when a fieldset has a checkbox field and none of the options are selected.

**Comment by billmn** [comment]
Even if you use code like this:

```
{{ collection:posts scope="post" }}
    {{ post:title }}

    {{ categories scope="category" }}
        {{ category:title }} <- this throw an exception
    {{ /categories }}
{{ /collection }}
```

"Posts" is a collection that have related "Categories" entries through an "Entries" field type in the blueprint. 

(I know that "scope" in this situation is not necessary but i think it shouldn't raise an exception if it is used)

**Comment by jimblue** [comment]
> Probably this happen even before the 3.0.5

It was perfectly working before 3.0.5

**Comment by jasonvarga** [comment]
Probably introduced in 3.0.4, so stay on 3.0.3 for the moment. Will fix this soon.

**Comment by edalzell** [comment]
I still get this error on 3.0.6. I've cleared all the caches, etc

**Comment by Pixney-William** [comment]
I get this issue with 3.0.6 

```
{{ paginate }}



{{if total_items > items_limit}}

<div class="w-full ">

    <div class="c-pagination">


        <a class="c-button" href="{{ prev_page }}">{{ svg src="left"}}</a>


        {{ current_page }} / {{ total_pages }}


        <a class="c-button" href="{{ next_page }}">{{ svg src="right"}}</a>

    </div>

</div>

{{ /if }}


{{ /paginate }}
```

**Comment by acurrieclark** [comment]
It seems there are lots of different possible types for `$values` here, not just `Value` or `Array`. I have a situation where with recursive navs where the `{{ depth }}` tag results in an `int` type.

**Comment by jasonvarga** [comment]
@acurrieclark Could you share your template where the depth variable is being used and causing the error?

**Comment by acurrieclark** [comment]
@jasonvarga Of course. Should have added it sooner.

https://gist.github.com/acurrieclark/a3e43a1e268cc35a980dd526029b175f

**Comment by jasonvarga** [comment]
Thanks @acurrieclark 

Looks like the cause is this line: https://gist.github.com/acurrieclark/a3e43a1e268cc35a980dd526029b175f#file-statamic-menu-php-L35

`{{ depth - 1 }}` is not something Antlers supports. Was this even doing what you expected before 3.0.4?

The way to do that in Antlers is to use the minus modifier. `{{ depth | -:1 }}`

**Comment by jimblue** [comment]
This is solved on my side with `3.0.6` 🙏🏼

**Comment by jasonvarga** [comment]
More fixes in 3.0.7
